src/Components/InMemoryEventStream.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Components
+{
+    internal sealed class InMemoryEventStream : EventStream, EventDispatcher
+    {
+        private readonly HashSet<object> _subscriptions = new HashSet<object>();
+
+        public void Dispatch<T>(T entity)
+        {
+            if (entity == null)
+                throw new ArgumentNullException(nameof(entity));
+
+            lock(_subscriptions)
+                foreach (var subscription in _subscriptions.OfType<IObserver<T>>())
+                    subscription.OnNext(entity);
+        }
+        public IDisposable Subscribe<T>(Action<T> callback)
+        {
+            if (callback == null)
+                throw new ArgumentNullException(nameof(callback));
+
+            return Subscribe(new SimpleSubscriber<T>(callback));
+        }
+        public IDisposable Subscribe<T>(IObserver<T> subscriber)
+        {
+            if (subscriber == null)
+                throw new ArgumentNullException(nameof(subscriber));
+
+            lock(_subscriptions)
+                _subscriptions.Add(subscriber);
+
+            return new Subscription(() => Unsubscribe(subscriber));
```

Comment:
Method Groups translate to Action's and Func's . You can pass any void parameterless method to something expecting an Action for example.